elementor/elementor
# Tweak: Add a VideoPress integration for self-hosted videos [ED-9068]

## PR Checklist



## PR Type
What kind of change does this PR introduce?


## Summary

This PR can be summarized in the following changelog entry:

* Use the VideoPress player for self-hosted VideoPress videos.
* Use the VideoPress player for self-hosted VideoPress external URLs.
* Fix a PHP warning when "lazy_load" is not in the embed options.

## Description

This PR creates a new video type for VideoPress videos in the Video widget.

The benefit to the user of using the VideoPress player over the default `<video>` tag is it is able to take advantage of all of the features that VideoPress provides such as using a highly compatible transcoded video file for playback on many types of devices, adaptive streaming based on device screen size and available bandwidth and the ability to authenticate private videos.

## Test instructions
This PR can be tested by following these steps:

* Apply this PR
* Create a new page or post
* Insert a video widget
* Select "VideoPress" as source
* Select "External URL"
* Set the URL to a public VideoPress URL, eg. `https://video.wordpress.com/embed/Ygmx4akX`
* ✅ The video should be properly loaded in the VideoPress player
* Test the same video GUID with different valid base URLs like:
    * `https://video.wordpress.com/v/Ygmx4akX`
    * `http://video.wordpress.com/v/Ygmx4akX`
    * `https://videopress.com/embed/Ygmx4akX`
    * `https://videopress.com/v/Ygmx4akX`
    * `http://videopress.com/v/Ygmx4akX`
    * `https://www.videopress.com/v/Ygmx4akX`
    * `videopress.com/v/Ygmx4akX`
    * `https://videos.files.wordpress.com/Ygmx4akX/red-line-1_mp4_std.mp4`
* ✅ All these URLs should load the VideoPress player with the correct video
* For the next step, two possibilites:
    * On a site with an active VideoPress subscription
    * Or Install Jetpack VideoPress and choose the free plan and make sure VideoPress is enabled
* Go to your media library and upload a video
* Copy the video URL starting with "videos.files.wordpress.com" or "video.wordpress.com" or "videopress.com"
* Go back to your post/page or create a new one
* Insert a video widget and select "VideoPress"
* Select the video you uploaded on VideoPress from your media library
* ✅ The VideoPress player should be displayed
* Select "external URL" and paste the URL you previously copied
* ✅ The VideoPress player should be displayed and the proper video should load.
* Make your video private by switching the "visibility" toggle in the media library
* Go back to your post/page
* ✅ The video should still work
* Publish your post/page
* Go to the frontend
* ✅ The video should still work
* Open an incognito tab and go to your page/post frontend
* ✅ The video should be displayed as private

## Quality assurance


Fix #20617
